pkg/cdi/cache.go

```diff
@@ -190,6 +191,41 @@ func (c *Cache) InjectDevices(ociSpec *oci.Spec, devices ...string) ([]string, e
 	return nil, nil
 }
 
+// WriteSpec writes a Spec file with the given content. Priority is used
+// as an index into the list of Spec directories to pick a directory for
+// the file, adjusting for any under- or overflows. If name has a "json"
+// or "yaml" extension it choses the encoding. Otherwise JSON encoding
+// is used with a "json" extension.
+func (c *Cache) WriteSpec(raw *cdi.Spec, name string) error {
+	var (
+		specDir string
+		path    string
+		prio    int
+		spec    *Spec
+		err     error
+	)
+
+	if len(c.specDirs) == 0 {
+		return errors.New("no Spec directories to write to")
+	}
+
+	prio = len(c.specDirs) - 1
```

Comment:
This logic seems very difficult to control. Do we have a use case where the caller doesn't just provide the full path?

Reply:
Well, the idea is that in the preferable default case the client does not need to know/care/try to control where the Spec file is spit out. This interface is for that. It always places the Spec file into the last configured Spec directory (IOW produces a file with the highest precedence) and overwrites any existing Spec file with the same name.

In case the client needs/wants to know/control, the lower-level `Spec.Write()` API is for that (which this interface also ends up using in the very end).

Reply:
OK. That make sense. I would update the doc comment to make this clearer though since the priority cannot be set by the user.